Feature/additional cypress tests

oftheheadland opened the pull request:
## Related Issues:
* https://app.breeze.pm/projects/100762/cards/3315171

## Main Changes:
* Adds tests for the Data, About, and ATTAINS pages that verify the pages load correctly and that the Back buttons work.

## Steps To Test:
1. Run Cypress tests for Data, About, and ATTAINS

cschwinderg commented on app/client/cypress/integration/attains/attains.spec.js:
```diff
@@ -0,0 +1,18 @@
+
+  it('ATTAINS Page Displays a Table', () => {
+    cy.visit('/attains');
+    cy.findByText("How's My Waterway Impairment Category").should('exist');
```
This test is failing for me because attains is running slow this morning. Can you add the following code to this test?

```
// wait for the all web services to finish (surveys is usually slow here)
cy.findAllByTestId('hmw-loading-spinner', { timeout: 20000 }).should('not.exist');
```

cschwinderg commented on app/client/cypress/integration/about/about.spec.js:
```diff
@@ -0,0 +1,15 @@
+    cy.findByText('About How’s My Waterway').should('exist');
+    cy.url().should('equal', `${document.location.origin}/about`);
+
+    cy.findByText('Back').click();
```
This test works well, but I think it should go a step further. One of the features with the back button is that the user can go back without having to reload anything. Can you modify this test to do a search, wait for loading spinners to go away, visit the about page, click the back button and then assert that something on the overview tab is exists?

cschwinderg replied:
Never mind, this one might not be a problem. It turns out that attains is completely down this morning.

cschwinderg commented on app/client/cypress/integration/data/data.spec.js:
```diff
@@ -0,0 +1,15 @@
+    cy.url().should('equal', `${document.location.origin}/data`);
+
+    cy.findByText('Back').click();
+    cy.findByText('Let’s get started!').should('exist');
```
Similar to the comment on about.spec.

oftheheadland commented:
@cschwinderg I've updated the About and Data tests to first search on the community page before testing the Back buttons so we can verify the content is still there.

I've also added some timeouts because the services have been slow and causing errors. Thanks!
